lib/ui/platform_dispatcher.dart

```diff
@@ -1251,6 +1296,106 @@ class WindowPadding {
   }
 }
 
+/// Area of the display that is obstructed by a hardware feature.
+///
+/// The bounds, which the application can use to guide layout, may be obscured
+/// or not have touch capabilities. The type can be used to determine behaviour.
+/// For example, a hinge feature can be used to separate the layout into 2
+/// logical areas or panels in the application.
+class DisplayFeature {
+  const DisplayFeature({
+    required this.rect,
+    required this.type,
+    required this.state,
+  });
+
+  DisplayFeature withDevicePixelRatio(double devicePixelRatio) {
+    return DisplayFeature(
+      rect : Rect.fromLTRB(
+        rect.left / devicePixelRatio,
+        rect.top / devicePixelRatio,
+        rect.right / devicePixelRatio,
+        rect.bottom / devicePixelRatio,
+      ),
+      type: type,
+      state: state,
+    );
+  }
+
+  /// Area of the view occupied by this display feature
+  final Rect rect;
+
+  /// Type of display feature, e.g. hinge, fold, cutout
+  final DisplayFeatureType type;
+
+  /// Posture of display feature, which is populated only for folds and hinges.
+  /// For cutouts, this is [DisplayFeatureState.unknown]
+  final DisplayFeatureState state;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is DisplayFeature &&
+          runtimeType == other.runtimeType &&
+          rect == other.rect &&
+          type == other.type &&
+          state == other.state;
+
+  @override
+  int get hashCode => hashValues(rect, type, state);
+
+  @override
+  String toString() {
+    return 'DisplayFeature(rect: $rect, type: $type, state: $state)';
+  }
+}
+
+/// Type of display feature (screen obstruction). This enum contains both
+/// foldable obstructions and cutout obstructions.
+///
+/// Some, like [DisplayFeatureType.fold], can be reported without actually
+/// impeding drawing on the screen. They are useful for knowing where the
+/// display is bent or has a crease. The [DisplayFeature] bounds can be
+/// 0-width in such cases.
+enum DisplayFeatureType {
+  unknown,
+  /// A fold in the flexible screen without a physical gap.
+  fold,
+  /// A physical separation with a hinge that allows two display panels to fold.
+  hinge,
+  /// A non-functional area of the screen, usually housing cameras or sensors.
+  cutout,
+}
+
+/// Display features for folds and hinges can have a state. Cutouts do not have
+/// a state.
+///
+///   * For cutouts, the state is [DisplayFeatureState.unknown].
+///   * For folds & hinges, the state is the posture.
+///   TODO: Once WM is stable, values should be, in this order: unknown, flat, halfOpened, flipped, with no other values.
+enum DisplayFeatureState {
```

Comment:
Question: is there a standard that is is based off of, or is this just new Flutter specific API?

Reply:
Never mind, I see this is based off of Android Postures. This should be noted in the overall enum docs.